src/backend/executor/nodeModifyTable.c

```diff
@@ -3809,6 +3809,12 @@ ExecModifyTable(PlanState *pstate)
 		if (pstate->ps_ExprContext)
 			ResetExprContext(pstate->ps_ExprContext);
 
+		if (check_rowcount_hook && check_rowcount_hook(estate->es_processed))
```

Comment:
this check will apply to even DMLs from PSQL dialect/endpoint, is that intended?